lib/presentation/screens/auth/register.dart

```diff
@@ -0,0 +1,154 @@
+
+  void _handleRegister() {
+    final email = _emailController.text.trim();
+    final username = _usernameController.text.trim();
+    final password = _passwordController.text;
+    final confirmPassword = _confirmPasswordController.text;
+
+    if ([email, username, password, confirmPassword]
+        .any((item) => item.isEmpty)) {
+      Get.snackbar(
+        "Registrasi Gagal",
+        "Semua data harus diisi",
+        snackPosition: SnackPosition.TOP,
+        colorText: Colors.white,
+      );
+      return;
+    }
+
+    if (password != confirmPassword) {
+      Get.snackbar(
+        "Registrasi Gagal",
+        "Kata sandi tidak valid",
+        snackPosition: SnackPosition.TOP,
+        colorText: Colors.white,
+      );
+      return;
+    }
+
+    Get.snackbar(
+      "Registrasi Berhasil",
+      "Akun berhasil didaftarkan",
+      snackPosition: SnackPosition.TOP,
+      colorText: Colors.white,
+    );
+
+    Get.toNamed('/signin');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BaseBackground(
+        child: Center(
+          child: SingleChildScrollView(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(vertical: 80, horizontal: 24),
+              child: Container(
+                constraints: const BoxConstraints(maxWidth: 400),
+                decoration: BoxDecoration(
+                  color: HexColor("0F1012").withOpacity(0.35),
+                  border: Border.all(
+                    color: Colors.white,
+                    width: 2,
+                  ),
+                  borderRadius: BorderRadius.circular(12),
+                ),
+                child: Padding(
+                  padding: const EdgeInsets.all(16),
+                  child: Center(
+                    child: Column(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        crossAxisAlignment: CrossAxisAlignment.center,
+                        children: [
+                          const Padding(
+                            padding: EdgeInsets.only(top: 16, bottom: 20),
+                            child: Text(
+                              'SIGN UP',
+                              style: TextStyle(
+                                  fontSize: 20, fontFamily: 'Poppins'),
+                            ),
+                          ),
+                          _RegisterForm(
+                              emailController: _emailController,
+                              usernameController: _usernameController,
+                              passwordController: _passwordController,
+                              confirmPasswordController:
+                                  _confirmPasswordController),
+                          const SizedBox(height: 20),
+                          button(label: "Daftar", onPressed: _handleRegister),
+                          const SizedBox(height: 28),
+                        ]),
```

Comment:
!!!!!!!!!!!!!!!!!!!!!!
```suggestion
                        ],),
```